main/src/ca/uwaterloo/flix/language/phase/Tailrec.scala

```diff
@@ -104,6 +104,8 @@ object Tailrec extends Phase[Root, Root] {
           Expression.ApplySelfTail(sym, defn.fparams, args, tpe, loc)
         }
 
+      //TODO SJ: can anything channel related be tail recursive
```

Comment:
Good question! Can you answer it yourself? :)

What do you think about:

```
def f(): select {
  case x <- if (..) f() else ..
}
```

?